cms/tests/test_templatetags.py

```diff
@@ -43,6 +43,48 @@
 
 
 class TemplatetagTests(CMSTestCase):
+    def test_get_admin_url_for_language(self):
+        # This creates a new page and the corresponding PageContent object for lang 'en', both with pk=1
+        page = create_page("AdminURLTestPage English Content", "nav_playground.html", "en")
+
+        # German PageContent will have pk=2
+        german_content = create_page_content("de", "AdminURLTestPage German Content", page)
+
+
+        request = RequestFactory().get('/')
+        request.current_page = page
+        with force_language('de'):
+            url = GetAdminUrlForLanguage.get_value(None, context={'request': request}, page=page, language="de")
+            # We expect that getting the German URL will return the German page content (change link, not a create link)
+            self.assertEqual(url, '/de/admin/cms/pagecontent/2/change/')
```

Comment:
```suggestion
            self.assertEqual(url, f'/de/admin/cms/pagecontent/{german_content.pk}/change/')
```
It's unsafe to make assumptions on pk's in testing environments.